cola/widgets/selectcommits.py

```diff
@@ -135,6 +142,13 @@ def commit_oid_double_clicked(self, _item):
         if oid:
             self.accept()
 
+    def search_list(self, text):
+        search_items = self.commits.findItems(text, Qt.MatchContains)
+        for i in range(self.commits.count()):
+            self.commits.item(i).setHidden(True)
+        for items in search_items:
+            items.setHidden(False)
```

Comment:
Hmm so the way it works is it first marks everything hidden and then only unhides the items that matched.
What if there are no matches? Do we need to special-case that and unhide everything?

Reply:
Hi no need for a special case as empty string matches contains everything which in result unhides everything. You can check in the demo GIF when i delete Pylint to enter typo everything is avaliable again.

Reply:
Maybe just skip the hidden check if we know everything is going to be unhidden anyways?

Reply:
Added some print statements for verification ->
Search time count for search  P  is :-  4536
Total item count is :-  7265
Search time count for search  Py  is :-  622
Total item count is :-  7265
Search time count for search  Pyl  is :-  174
Total item count is :-  7265
Search time count for search  Pyli  is :-  174
Total item count is :-  7265
Search time count for search  Pylin  is :-  174
Total item count is :-  7265
Search time count for search  Pylint  is :-  174
Total item count is :-  7265
Search time count for search  Pylin  is :-  174
Total item count is :-  7265
Search time count for search  Pyli  is :-  174
Total item count is :-  7265
Search time count for search  Pyl  is :-  174
Total item count is :-  7265
Search time count for search  Py  is :-  622
Total item count is :-  7265
Search time count for search  P  is :-  4536
Total item count is :-  7265
Search time count for search    is :-  7265 <-------------------------------Check this line when i cleared the search box all items show up.
Total item count is :-  7265
Search time count for search  T  is :-  7013
Total item count is :-  7265
Search time count for search  Ty  is :-  335
Total item count is :-  7265
Search time count for search  Tyo  is :-  0
Total item count is :-  7265
Search time count for search  Ty  is :-  335
Total item count is :-  7265
Search time count for search  Typ  is :-  86
Total item count is :-  7265
Search time count for search  Typu  is :-  0
Total item count is :-  7265
Search time count for search  Typ  is :-  86
Total item count is :-  7265
Search time count for search  Typo  is :-  48
Total item count is :-  7265

Reply:
Would be interesting to measure the time on some huge git repos like the Linux kernel...

Reply:
@kakra You mean is text is empty don't hide it ? Good idea ! let me try.